File: forest_contest/06/06-02.py
# 最大イニング (paizaランク D 相当)
# https://paiza.jp/works/mondai/forest_contest_006/forest_contest_006__d_maxinning

INPUT1 = """\
99
6
16
14
22
8
11
7
10
4"""
OUTPUT1 = "Yes"

INPUT2 = """\
70
6
16
14
22
8
11
7
10
4"""

OUTPUT2 = "6"

INPUT3 = """\
77
6
16
14
22
8
11
7
10
4"""
OUTPUT3 = "7"

# 解答例1-1
n = 9
k = int(input())
a = [ int(input()) for _ in range(n)]

leaving_inning = None   # 交代イニング
num_of_pitches = 0      # 現在の投球数
for i in range(n):
    num_of_pitches += a[i]

    # k 球を超えた?
    if num_of_pitches > k:
        leaving_inning = i + 1
        break

if leaving_inning is None:
    print("Yes")
else:
    print(leaving_inning)

# 解答例1-2
k, *a = map(int, open(0).read().strip().split())

leaving_inning = None   # 交代イニング
num_of_pitches = 0      # 現在の投球数
for (i, v) in enumerate(a, 1):
    num_of_pitches += v

    # k 球を超えた?
    if num_of_pitches > k:
        leaving_inning = i
        break

print(leaving_inning or "Yes")

File: forest_contest/06/06-02.rb
# 最大イニング (paizaランク D 相当)
# https://paiza.jp/works/mondai/forest_contest_006/forest_contest_006__d_maxinning

INPUT1 = <<"EOS"
99
6
16
14
22
8
11
7
10
4
EOS
OUTPUT1 = <<"EOS"
Yes
EOS

INPUT2 = <<"EOS"
70
6
16
14
22
8
11
7
10
4
EOS
OUTPUT2 = <<"EOS"
6
EOS

INPUT3 = <<"EOS"
77
6
16
14
22
8
11
7
10
4
EOS
OUTPUT3 = <<"EOS"
7
EOS

k,*a=$<.map &:to_i;l=nil;n=0;(0..8).each{|i|n+=a[i];l=i+1 if n>k;break if l};puts l||"Yes"

=begin
# 解答例1-1
n = 9
k = gets.to_i
a = n.times.map { gets.to_i }

leaving_inning = nil  # 交代イニング
num_of_pitches = 0    # 現在の投球数
0.upto(n - 1) do |i|
  num_of_pitches += a[i]

  # k 球を超えた?
  if num_of_pitches > k
    leaving_inning = i + 1
    break
  end
end

if leaving_inning.nil?
  puts "Yes"
else
  puts leaving_inning
end

# 解答例1-2
k, *a = $stdin.read.split.map(&:to_i)

leaving_inning = nil  # 交代イニング
num_of_pitches = 0    # 現在の投球数
a.each.with_index(1) do |v, i|
  num_of_pitches += v

  # k 球を超えた?
  if num_of_pitches > k
    leaving_inning = i
    break
  end
end

puts leaving_inning || "Yes"
=end